### DISCUSSION TITLE
"A stop job is running" problem

### DISCUSSION THREAD

**Discussion by udok29** [gerbera/gerbera#2813] [Q&A]
Hello,
when shutting down my computer I have a problem with gerbera. The shutdown is delayed by gerbera. The message "**A Stop job is running for Gerbera Media Server with MysQL (42s / 1min 30s)**" appears.
I can't find the cause. I suspect it is related to MysQL. I don't have MysQL installed. Sqlite3 is running on my computer.
I noticed another (error) message when manually stopping and starting gerbera with systemctl. Gerbera tries to upgrade a database but fails.

The following is entered in the config.xml file:
```
     ........<storage>
       <sqlite3 enabled="yes">
         <database-file>/home/udo/.config/gerbera/gerbera.db</database-file>
       </sqlite3>
       <mysql enabled="no">
         <host>localhost</host>
         <username>gerbera</username>
         <database>gerbera</database>
       </mysql>
     </storage>........
```
     
If Gerbera is started and I call "gerbera" in a console, the films are streamed in the home network. I have attached an extract from the log files.
Can you please help me?

Operating System: Kubuntu 22.04
KDE Plasma Version: 5.24.7
KDE Frameworks Version: 5.92.0
Qt Version: 5.15.3
Kernel Version: 5.15.0-69-generic (64-bit)
Graphics Platform: X11
Processors: 16 × Intel® Xeon® CPU E5-2620 v4 @ 2.10GHz
Memory: 31.3 GiB of RAM
Graphics Processor: NVIDIA GeForce GTX 1060 6GB/PCIe/SSE2

#**cat /usr/lib/systemd/system/gerbera.service**
```
[Unit]
Description=Gerbera Media Server with MySQL
After=mysql.target network-online.target
Wants=network-online.target

[Service]
Type=simple
User=gerbera
Group=gerbera
ExecStart=/usr/bin/gerbera -c /etc/gerbera/config.xml
Restart=on-failure
RestartSec=5

[Install]
WantedBy=multi-user.target
```

**Notice:** Remove "mysql.target" don't help.

**#systemctl status gerbera**
```
● gerbera.service - Gerbera Media Server with MySQL
     Loaded: loaded (/lib/systemd/system/gerbera.service; enabled; vendor preset: enabled)
     Active: active (running) since Sun 2023-04-02 14:18:41 CEST; 4h 52min ago
   Main PID: 1933 (gerbera)
      Tasks: 18 (limit: 38225)
     Memory: 11.2M
        CPU: 1.967s
     CGroup: /system.slice/gerbera.service
             └─1933 /usr/bin/gerbera -c /etc/gerbera/config.xml

Apr 02 14:18:41 pc08 systemd[1]: Started Gerbera Media Server with MySQL.
Apr 02 14:18:41 pc08 gerbera[1933]: 2023-04-02 14:18:41   info: Gerbera UPnP Server 1.12.1 - http://gerbera.io/
Apr 02 14:18:41 pc08 gerbera[1933]: 2023-04-02 14:18:41   info: Copyright 2016-2022 Gerbera Contributors.
Apr 02 14:18:41 pc08 gerbera[1933]: 2023-04-02 14:18:41   info: Gerbera is free software, covered by the GNU Gen>
Apr 02 14:18:41 pc08 gerbera[1933]: 2023-04-02 14:18:41   info: Loading configuration from: /etc/gerbera/config.>
Apr 02 14:18:41 pc08 gerbera[1933]: 2023-04-02 14:18:41   info: Checking configuration...
Apr 02 14:18:41 pc08 gerbera[1933]: 2023-04-02 14:18:41   info: Configuration check succeeded.
Apr 02 14:18:41 pc08 gerbera[1933]: 2023-04-02 14:18:41   info: Running an automatic database upgrade from datab>
Apr 02 14:18:41 pc08 gerbera[1933]: 2023-04-02 14:18:41  error: prematurely shutting down.

```
**syslog**:
syslog.1:Mar 24 **23:03:37** pc08 systemd[1]: **Stopping** Gerbera Media Server with MySQL...
syslog.1:Mar 24 **23:05:08** pc08 systemd[1]: gerbera.service: State 'stop-sigterm' timed out. Killing.
syslog.1:Mar 24 23:05:08 pc08 systemd[1]: gerbera.service: Killing process 1811 (gerbera) with signal SIGKILL.
syslog.1:Mar 24 23:05:08 pc08 systemd[1]: gerbera.service: Killing process 1823 (gerbera) with signal SIGKILL.
syslog.1:Mar 24 23:05:08 pc08 systemd[1]: gerbera.service: Killing process 1825 (gerbera) with signal SIGKILL.
syslog.1:Mar 24 23:05:08 pc08 systemd[1]: gerbera.service: Killing process 1826 (n/a) with signal SIGKILL.
syslog.1:Mar 24 23:05:08 pc08 systemd[1]: gerbera.service: Killing process 1828 (n/a) with signal SIGKILL.
syslog.1:Mar 24 23:05:08 pc08 systemd[1]: gerbera.service: Main process exited, code=killed, status=9/KILL
syslog.1:Mar 24 23:05:08 pc08 systemd[1]: gerbera.service: Failed with result 'timeout'.
syslog.1:Mar 24 23:05:08 pc08 systemd[1]: Stopped Gerbera Media Server with MySQL.
syslog.1:Mar 24 23:05:08 pc08 systemd[1]: gerbera.service: Consumed 2.009s CPU time.
syslog.1:Mar 25 12:22:54 pc08 systemd[1]: Started Gerbera Media Server with MySQL.
syslog.1:Mar 25 12:22:54 pc08 gerbera[1655]: 2023-03-25 12:22:54   info: Gerbera UPnP Server 1.12.1 - http://gerbera.io/
syslog.1:Mar 25 12:22:54 pc08 gerbera[1655]: 2023-03-25 12:22:54   info: Copyright 2016-2022 Gerbera Contributors.
syslog.1:Mar 25 12:22:54 pc08 gerbera[1655]: 2023-03-25 12:22:54   info: Gerbera is free software, covered by the GNU General Public License version 2
syslog.1:Mar 25 12:22:54 pc08 gerbera[1655]: 2023-03-25 12:22:54   info: Loading configuration from: /etc/gerbera/config.xml
syslog.1:Mar 25 12:22:54 pc08 gerbera[1655]: 2023-03-25 12:22:54   info: Checking configuration...
syslog.1:Mar 25 12:22:54 pc08 gerbera[1655]: 2023-03-25 12:22:54   info: Configuration check succeeded.
syslog.1:Mar 25 12:22:54 pc08 gerbera[1655]: 2023-03-25 12:22:54   info: **Running an automatic database upgrade from database version 12 to version 13...**
syslog.1:Mar 25 12:22:54 pc08 gerbera[1655]: 2023-03-25 12:22:54  **error**: prematurely shutting down.
syslog.1:Mar 25 12:22:54 pc08 gerbera[1655]: 2023-03-25 12:22:54  **error**: [/build/src/database/sqlite3/sqlite_database.cc:171] virtual void Sqlite3Database::init() Error: [/build/src/database/sqlite3/sqlite_database.cc:313] virtual int Sqlite3Database::exec(const string&, bool) Error: [/build/src/database/sqlite3/sl_task.cc:61] void SLTask::waitForTask() Error: SQLITE3 (1: 1): table "grb_cds_resource" already exists
syslog.1:Mar 25 12:22:54 pc08 gerbera[1655]:        Query: CREATE TABLE "grb_cds_resource" (
syslog.1:Mar 25 12:22:54 pc08 gerbera[1655]:             "id" integer primary key,
syslog.1:Mar 25 12:22:54 pc08 gerbera[1655]:             "item_id" integer NOT NULL,
syslog.1:Mar 25 12:22:54 pc08 gerbera[1655]:             "res_id" integer NOT NULL,
syslog.1:Mar 25 12:22:54 pc08 gerbera[1655]:             "handlerType" integer NOT NULL,
syslog.1:Mar 25 12:22:54 pc08 gerbera[1655]:             "options" text default NULL,
syslog.1:Mar 25 12:22:54 pc08 gerbera[1655]:             "parameters" text default NULL,
syslog.1:Mar 25 12:22:54 pc08 gerbera[1655]:             CONSTRAINT "grb_cds_resource_fk" FOREIGN KEY ("item_id") REFERENCES "mt_cds_object" ("id") ON DELETE CASCADE ON UPDATE CASCADE
syslog.1:Mar 25 12:22:54 pc08 gerbera[1655]:         );
syslog.1:Mar 25 12:22:54 pc08 gerbera[1655]:         INSERT INTO "mt_internal_setting" VALUES('resource_attribute', '');
syslog.1:Mar 25 12:22:54 pc08 gerbera[1655]:         CREATE INDEX grb_cds_resource_id ON grb_cds_resource(item_id,res_id);
syslog.1:Mar 25 12:22:54 pc08 gerbera[1655]:        Error: table "grb_cds_resource" already exists
syslog.1:Mar 25 12:22:54 pc08 systemd[1]: gerbera.service: Main process exited, code=exited, status=1/FAILURE
syslog.1:Mar 25 12:22:54 pc08 systemd[1]: gerbera.service: Failed with result 'exit-code'.
syslog.1:Mar 25 12:22:54 pc08 systemd[1]: gerbera.service: Consumed 1.300s CPU time.
syslog.1:Mar 25 12:22:59 pc08 systemd[1]: gerbera.service: Scheduled restart job, restart counter is at 1.
syslog.1:Mar 25 12:22:59 pc08 systemd[1]: Stopped Gerbera Media Server with MySQL.
syslog.1:Mar 25 12:22:59 pc08 systemd[1]: gerbera.service: Consumed 1.300s CPU time.
syslog.1:Mar 25 12:22:59 pc08 systemd[1]: Started Gerbera Media Server with MySQL.
syslog.1:Mar 25 12:22:59 pc08 gerbera[1831]: 2023-03-25 12:22:59   info: Gerbera UPnP Server 1.12.1 - http://gerbera.io/
syslog.1:Mar 25 12:22:59 pc08 gerbera[1831]: 2023-03-25 12:22:59   info: Copyright 2016-2022 Gerbera Contributors.
syslog.1:Mar 25 12:22:59 pc08 gerbera[1831]: 2023-03-25 12:22:59   info: Gerbera is free software, covered by the GNU General Public License version 2
syslog.1:Mar 25 12:22:59 pc08 gerbera[1831]: 2023-03-25 12:22:59   info: Loading configuration from: /etc/gerbera/config.xml
syslog.1:Mar 25 12:22:59 pc08 gerbera[1831]: 2023-03-25 12:22:59   info: Checking configuration...
syslog.1:Mar 25 12:22:59 pc08 gerbera[1831]: 2023-03-25 12:22:59   info: Configuration check succeeded.
syslog.1:Mar 25 12:22:59 pc08 gerbera[1831]: 2023-03-25 12:22:59   info: Running an automatic database upgrade from database version 12 to version 13...
syslog.1:Mar 25 12:22:59 pc08 gerbera[1831]: 2023-03-25 12:22:59  error: prematurely shutting down.

**dpkg.log**:2023-03-27 19:27:45 remove gerbera:amd64 1.12.1-jammy1 <none>
dpkg.log:2023-03-27 19:27:45 status half-configured gerbera:amd64 1.12.1-jammy1
dpkg.log:2023-03-27 19:27:45 status half-installed gerbera:amd64 1.12.1-jammy1
dpkg.log:2023-03-27 19:27:45 status config-files gerbera:amd64 1.12.1-jammy1
dpkg.log:2023-03-27 19:29:02 install gerbera:amd64 1.12.1-jammy1 1.12.1-jammy1
dpkg.log:2023-03-27 19:29:02 status half-installed gerbera:amd64 1.12.1-jammy1
dpkg.log:2023-03-27 19:29:02 status unpacked gerbera:amd64 1.12.1-jammy1
dpkg.log:2023-03-27 19:29:02 configure gerbera:amd64 1.12.1-jammy1 <none>
dpkg.log:2023-03-27 19:29:02 status unpacked gerbera:amd64 1.12.1-jammy1
dpkg.log:2023-03-27 19:29:02 status half-configured gerbera:amd64 1.12.1-jammy1
dpkg.log:2023-03-27 19:29:02 status installed gerbera:amd64 1.12.1-jammy1

**Comment by KarlStraussberger** [comment]
Seems that you updated from a really old version.

I suggest to delete the database file (and the backup). Your media will be imported again which will give you all the new features.

**Reply by udok29** [reply]
Thank you, that helped.
What have I done in detail?
- apt purge gerbera
- Delete all directories and files belonging to gerbera: `find / -name \*gerbera\* , rmdir and rm`
- Reinstallation: apt install gerbera
- Set up as a daemon using the documentation.

Documentation notes:
After installing gerbera, the file gerbera.service is in the directory "/**usr/lib**/systemd/system/gerbera.service" and not under "/etc....."
In order for gerbera to start the next time the computer is booted, "**systemctl enable gerbera**" must be executed. This will create a link --> "/etc/systemd/system/multi-user.target.wants/gerbera.service" that will make this happen

gerbera.service : If MySQL is not installed, the entries are irritating. Maybe this is customizable.
